inc/class-site-meta.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ * Class for fetching site information as used by Bluehost.
+ *
+ * @package BluehostWordPressPlugin
+ */
+
+/**
+ * Class Bluehost_Site_Meta
+ */
+class Bluehost_Site_Meta {
+
+	/**
+	 * Get the domain for the WordPress installation.
+	 *
+	 * @return string
+	 */
+	public static function get_domain() {
+		return wp_parse_url( get_home_url(), PHP_URL_HOST );
+	}
+
+	/**
+	 * Get the relative filesystem path to the root of the WordPress installation.
+	 *
+	 * @return string
+	 */
+	public static function get_path() {
+		return strstr( ABSPATH, '/public_html/' );
+	}
+
+	/**
+	 * Get the (Bluehost) site ID given the full path to the site root.
+	 *
+	 * @return string Site ID is the path encoded as UTF-8 and converted to hexadecimal.
+	 */
+	public static function get_id() {
+		$path = self::get_path();
+
+		return bin2hex( iconv( mb_detect_encoding( $path, mb_detect_order(), true ), 'UTF-8', $path ) );
```

Comment:
Is this something we want to crunch on each use? Does `ABSPATH` change if a site was pushed to a new box..? 

Not that there's a huge expense in running this logic, but I feel like having this ID in an option would be helpful for purging cache and access from CLI so we could still use `--skip-plugins` in internal calls.

Reply:
@0aveRyan I don't think we'd need to use the ID outside of the plugin. This is most likely only going to be needed for API calls we make or validating REST requests from the other side. At the moment, this is only ever called once during runtime every hour. Adding it to an option would probably be more overhead at the moment. However, we could use a static variable in this method just in case it is called more than once during runtime. What do you think?

Reply:
Updated function to use a static variable for caching. 

For documentation of this decision: Since the ID doesn't really change unless the site is relocated, there is no `save` method. As such, saving to the database would require a lot more logic including fetching the value and comparing to see if it should be updated... at which point it would defeat the purpose of adding the value to the database.